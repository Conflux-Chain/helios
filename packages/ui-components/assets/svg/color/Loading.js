function SvgLoading(props) {
  return (
    <svg
      width={52}
      height={52}
      viewBox="0 0 52 52"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <circle cx={26} cy={26} r={24} stroke="currentColor" strokeWidth={3} />
      <path
        d="M26 50c13.255 0 24-10.745 24-24S39.255 2 26 2"
        stroke="#44D7B6"
        strokeWidth={3}
      />
    </svg>
  )
}

export default SvgLoading
