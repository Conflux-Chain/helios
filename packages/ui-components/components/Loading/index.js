import PropTypes from 'prop-types'
import {Loading as LoadingSvg} from '../../assets/svg'

function Loading({className = ''}) {
  return (
    <LoadingSvg
      className={`animate-spin text-gray-20 w-12 h-12 ${className}`}
    />
  )
}

export default Loading

Loading.propTypes = {
  className: PropTypes.string,
}
