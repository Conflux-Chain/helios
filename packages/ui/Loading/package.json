{
  "version": "0.0.0",
  "name": "@cfxjs/loading",
  "type": "module",
  "main": "index.js"
}
