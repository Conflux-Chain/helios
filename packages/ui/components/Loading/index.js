import React from 'react'
import PropTypes from 'prop-types'
import LoadingSvg from './assets/Loading'

function Loading({className = ''}) {
  return (
    <LoadingSvg
      data-testid="loading-wrapper"
      className={`animate-spin text-gray-20 w-12 h-12 ${className}`}
    />
  )
}

export default Loading

Loading.propTypes = {
  className: PropTypes.string,
}
