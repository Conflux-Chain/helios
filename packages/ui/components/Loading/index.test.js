import {render, screen} from '@testing-library/react'
import Loading from './index.js'
import {describe, expect} from '@jest/globals'

describe('Loading', () => {
  it('test snapshot', () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const snapshot = render(<Loading className="test-loading" />)
    expect(snapshot).toMatchSnapshot()
  })
  it('should render incoming class name', () => {
    render(<Loading className="test-loading" />)
    expect(screen.getByTestId('loading-wrapper')).toHaveClass('test-loading')
  })
})
