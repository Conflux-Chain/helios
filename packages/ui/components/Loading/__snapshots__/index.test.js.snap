// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading test snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        class="animate-spin text-gray-20 w-12 h-12 test-loading"
        data-testid="loading-wrapper"
        fill="none"
        height="52"
        viewBox="0 0 52 52"
        width="52"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="26"
          cy="26"
          r="24"
          stroke="currentColor"
          stroke-width="3"
        />
        <path
          d="M26 50c13.255 0 24-10.745 24-24S39.255 2 26 2"
          stroke="#6FC5B1"
          stroke-width="3"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="animate-spin text-gray-20 w-12 h-12 test-loading"
      data-testid="loading-wrapper"
      fill="none"
      height="52"
      viewBox="0 0 52 52"
      width="52"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="26"
        cy="26"
        r="24"
        stroke="currentColor"
        stroke-width="3"
      />
      <path
        d="M26 50c13.255 0 24-10.745 24-24S39.255 2 26 2"
        stroke="#6FC5B1"
        stroke-width="3"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
